src/wmtk/Tuple.hpp

```diff
@@ -78,6 +78,10 @@ class Tuple
     /// Checks if a tuple is "null". This merely implies the global index is -1
     bool is_null() const;
     Tuple with_updated_hash(int64_t new_hash) const;
+
+    int8_t local_vid() const;
```

Comment:
IIRC we purposely were leaving these as implementation details - please don't make public

Reply:
(just to increase the friction of most people using these to do things, like access topological attributes directly)

Reply:
I made the component a friend of Tuple and made the getters private.